### PR TITLE
Introspection of HasTraits classes

### PR DESCRIPTION
**Pull request by denistex**
Hi guys!

This is PR for #9000. I'm not sure if introspection output looks good enough (it's not colored and hard to parse visually), so check it please. How do you think?

### CODE REVIEW

**Comment by takluyver** [comment]
I'm not sure that we should special case a specific object system (traitlets) in our introspection machinery. I'll try to think about better protocols for complex dynamic objects to give us more info about themselves.

**Comment by denis-it** [comment]
Great idea!) So should I close this PR?

**Comment by takluyver** [comment]
This is now quite old, and I don't think special casing traitlets is the right way forwards. So I'm going to close this. We can use #9000 for further discussion of how to tackle it.
